cas-management-webapp/src/main/java/org/jasig/cas/services/web/beans/RegisteredServiceEditBean.java

```diff
@@ -537,82 +849,14 @@ public RegisteredService toRegisteredService(final IPersonAttributeDao dao) {
-                    }
-                }
-                ((DefaultRegisteredServiceAccessStrategy) accessStrategy)
-                        .setRequiredAttributes(requiredAttrs);
-
-                final String proxyType = this.proxyPolicy.getType();
-                if (StringUtils.equalsIgnoreCase(proxyType,
-                        RegisteredServiceProxyPolicyBean.Types.REFUSE.toString())) {
-                    regSvc.setProxyPolicy(new RefuseRegisteredServiceProxyPolicy());
-                } else if (StringUtils.equalsIgnoreCase(proxyType,
-                        RegisteredServiceProxyPolicyBean.Types.REGEX.toString())) {
-                    final String value = this.proxyPolicy.getValue();
-                    if (StringUtils.isNotBlank(value) && isValidRegex(value)) {
-                        regSvc.setProxyPolicy(new RegexMatchingRegisteredServiceProxyPolicy(value));
-                    } else {
-                        throw new IllegalArgumentException("Invalid regex pattern specified for proxy policy: " + value);
-                    }
-                }
+                setProxyPolicyOnService(regSvc);
 
-                final String uidType = this.userAttrProvider.getType();
-                if (StringUtils.equalsIgnoreCase(uidType,
-                        RegisteredServiceUsernameAttributeProviderEditBean.Types.DEFAULT.toString())) {
-                    regSvc.setUsernameAttributeProvider(new DefaultRegisteredServiceUsernameProvider());
-                } else if (StringUtils.equalsIgnoreCase(uidType,
-                        RegisteredServiceUsernameAttributeProviderEditBean.Types.ANONYMOUS.toString())) {
-                    final String salt = this.userAttrProvider.getValue();
-                    if (StringUtils.isNotBlank(salt)) {
-
-                        final ShibbolethCompatiblePersistentIdGenerator generator =
-                                new ShibbolethCompatiblePersistentIdGenerator(salt);
-                        regSvc.setUsernameAttributeProvider(
-                                new AnonymousRegisteredServiceUsernameAttributeProvider(generator));
-                    } else {
-                        throw new IllegalArgumentException("Invalid sale value for anonymous ids " + salt);
-                    }
-                } else if (StringUtils.equalsIgnoreCase(uidType,
-                        RegisteredServiceUsernameAttributeProviderEditBean.Types.ATTRIBUTE.toString())) {
-                    final String attr = this.userAttrProvider.getValue();
-
-                    if (StringUtils.isNotBlank(attr)) {
-                        regSvc.setUsernameAttributeProvider(
-                                new PrincipalAttributeRegisteredServiceUsernameProvider(attr));
-                    } else {
-                        throw new IllegalArgumentException("Invalid attribute specified for username");
-                    }
-                }
+                setUsernameAttributeOnService(regSvc);
```

Comment:
This is relevant to Checkstyle updates for reducing CC. The code is refactored into a number of smaller methods, is all. No other changes.